ts/components/services/ContactPreferencesToggles/index.tsx

```diff
@@ -1,91 +1,169 @@
-import React, { useState } from "react";
+import * as pot from "italia-ts-commons/lib/pot";
+import React, { useEffect, useState } from "react";
 import { connect } from "react-redux";
 import { fromNullable } from "fp-ts/lib/Option";
-
 import { GlobalState } from "../../../store/reducers/types";
 import I18n from "../../../i18n";
 import ItemSeparatorComponent from "../../ItemSeparatorComponent";
 import { NotificationChannelEnum } from "../../../../definitions/backend/NotificationChannel";
+import { Dispatch } from "../../../store/actions/types";
+import {
+  servicePreferenceSelector,
+  ServicePreferenceState
+} from "../../../store/reducers/entities/services/servicePreference";
+import { ServiceId } from "../../../../definitions/backend/ServiceId";
+import {
+  loadServicePreference,
+  upsertServicePreference
+} from "../../../store/actions/services/servicePreference";
+import { serviceIDCurrentSelector } from "../../../store/reducers/entities/services/currentService";
+import {
+  isServicePreferenceResponseSuccess,
+  ServicePreference
+} from "../../../types/services/ServicePreferenceResponse";
+import { isStrictSome } from "../../../utils/pot";
+import { showToast } from "../../../utils/showToast";
 import PreferenceToggleRow from "./PreferenceToggleRow";
 
 type Item = "email" | "push" | "inbox";
 
 type Props = {
   channels?: ReadonlyArray<NotificationChannelEnum>;
-} & ReturnType<typeof mapStateToProps>;
+} & ReturnType<typeof mapStateToProps> &
+  ReturnType<typeof mapDispatchToProps>;
+
+const hasChannel = (
+  channel: NotificationChannelEnum,
+  channels?: ReadonlyArray<NotificationChannelEnum>
+) =>
+  fromNullable(channels)
+    .map(anc => anc.indexOf(channel) !== -1)
+    .getOrElse(true);
+
+const getValueOrFalse = (
+  potServicePreference: ServicePreferenceState,
+  key: Item
+): boolean => {
```

Comment:
```suggestion
const isChanelEnabled = (
  potServicePreference: ServicePreferenceState,
  key: Item
): boolean => {
```
`getValueOrFalse` doesn't speak about what means the value returned

Reply:
@Undermaken maybe `isChannelEnabled` could be confused with the available channels... what do you think of this?

```suggestion
/**
 * Utility function to get the user preference value for a specific channel
 * return false if preference state is pot.none or if an error occurred on API Response
**/
const getChannelPreference = (
  potServicePreference: ServicePreferenceState,
  key: Item
): boolean => {
```